assets/app/scripts/directives/fromFile.js

```diff
@@ -0,0 +1,318 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .directive("fromFile", function($q,
+                                  $uibModal,
+                                  $location,
+                                  $filter,
+                                  CachedTemplateService,
+                                  Navigate,
+                                  TaskList,
+                                  DataService,
+                                  APIService) {
+    return {
+      restrict: "E",
+      scope: false,
+      templateUrl: "views/directives/from-file.html",
+      controller: function($scope) {
+        var aceEditorSession;
+        var humanize = $filter('humanize');
+        var humanizeKind = $filter('humanizeKind');
+        TaskList.clear();
+
+        $scope.aceLoaded = function(editor) {
+          aceEditorSession = editor.getSession();
+          aceEditorSession.setOption('tabSize', 2);
+          aceEditorSession.setOption('useSoftTabs', true);
+          editor.setDragDelay = 0;
+          editor.$blockScrolling = Infinity;
+
+          $('#from-file .editor').animate({
+            height: Math.floor(window.innerHeight * 0.50) + 'px'
+          }, 30);
```

Comment:
So we're making it half the window height? What it my window is not maximized or I'm on a phone. I'd rather just give it a fixed height.